interoperator/controllers/schedulers/sflabelselectorscheduler/sflabelselectorscheduler_controller.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2018 The Service Fabrik Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sflabelselectorscheduler
+
+import (
+	"context"
+	"math"
+	"os"
+	"strings"
+
+	osbv1alpha1 "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api/osb/v1alpha1"
+
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/internal/config"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/internal/dynamic"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/internal/renderer/gotemplate"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/cluster/registry"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/errors"
+
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SFLabelSelectorScheduler reconciles a SFLabelSelectorScheduler object
+type SFLabelSelectorScheduler struct {
+	client.Client
+	Log             logr.Logger
+	scheme          *runtime.Scheme
+	clusterRegistry registry.ClusterRegistry
+}
+
+// Reconcile schedules the SFServiceInstance to one SFCluster and sets the ClusterID in
+// SFServiceInstance.Spec.ClusterID. It chooses the cluster with least number of
+// SFServiceInstances already deployed
+func (r *SFLabelSelectorScheduler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("sfserviceinstance", req.NamespacedName)
+
+	instance := &osbv1alpha1.SFServiceInstance{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	if instance.Spec.ClusterID == "" {
+		labelSelector, err := getLabelSelectorString(*instance, *r)
+		if err != nil || labelSelector == "" {
```

Comment:
Should we fail for empty label selector as well ? I guess no.

Reply:
Done